File: lib/models/SanPham.dart
class SanPham {
  SanPham({
    required this.sanPhamId,
    required this.tenSanPham,
    required this.anhSanPham,
    required this.giaBan,
    required this.soLuongTon,
    required this.danhMucId,
    required this.moTa,
  });

  final int? sanPhamId;
  final String? tenSanPham;
  final String? anhSanPham;
  final int? giaBan;
  final int? soLuongTon;
  final int? danhMucId;
  final String? moTa;

  factory SanPham.fromJson(Map<String, dynamic> json) {
    return SanPham(
      sanPhamId: json["sanPhamId"],
      tenSanPham: json["tenSanPham"],
      anhSanPham: json["anhSanPham"],
      giaBan: json["giaBan"],
      soLuongTon: json["soLuongTon"],
      danhMucId: json["danhMucId"],
      moTa: json["moTa"],
    );
  }
}

File: API_DBFruits/API_DBFruits/Models/SanPham.cs
using System;
using System.Collections.Generic;

namespace API_DBFruits.Models;

public partial class SanPham
{
    public int SanPhamId { get; set; }

    public string TenSanPham { get; set; } = null!;

    public string AnhSanPham { get; set; } = null!;

    public int GiaBan { get; set; }

    public int SoLuongTon { get; set; }

    public int? DanhMucId { get; set; }

    public string? MoTa { get; set; }

    public virtual ICollection<ChiTietDonHang> ChiTietDonHangs { get; set; } = new List<ChiTietDonHang>();

    public virtual ICollection<DanhMucYeuThich> DanhMucYeuThiches { get; set; } = new List<DanhMucYeuThich>();
    public virtual ICollection<GioHang> GioHangs { get; set; } = new List<GioHang>();

    public virtual DanhMucSanPham? DanhMuc { get; set; }
}
